src/accelerator.rs

```diff
@@ -0,0 +1,98 @@
+use drivers::Driver;
+use Motion;
+use Operation;
+
+pub struct Accelerator<D: Driver> {
+    motions: Vec<Motion>,
+    driver: D,
+}
+
+impl <D: Driver> Accelerator<D> {
+    pub fn new(mots: Vec<Motion>, drv: D) -> Self {
+        Accelerator{motions: mots, driver: drv}
+    }
+    // Needs to be un pub.ed
+    pub fn within_bounds(&self, n: isize) -> isize{
+        println!("called bounds with \'{}\'", n);
+        let min = 0;
+        let max = self.motions.len() as isize;
+        if n < min { return min }
+        if n > max { return max }
+        n
+    }
+
+    fn execute(&mut self, mut start: isize, mut finish: isize){
+        println!("called execute with \'{0}\' and \'{1}\'", start,finish);
+        if start != finish {
+            start = self.within_bounds(start);
+            finish = self.within_bounds(finish);
+            let operation = Operation::get(finish, start);
+            match operation{
+                Operation::Add(op) => {
+                    let mut i = start;
+                    loop{
+                        if i == finish { break }
+                        self.driver.execute(self.motions[i as usize].add);
+                        i += op;
+                    }
+                    self.driver.set_status(i);
+                },
+                Operation::Sub(op) => {
+                    let mut i = start;
+                    loop{
+                        if i == finish { break }
+                        i += op;
```

Comment:
yes because I hate it.. but the shift wouldn't produce the correct result thus this was the best fix I could come up with... I'm still trying to think of a way of not having it be this way.